engine/runtime-parser/src/main/java/org/enso/compiler/core/ir/IrPersistance.java

```diff
@@ -0,0 +1,760 @@
+package org.enso.compiler.core.ir;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+
+import org.enso.compiler.core.CompilerStub;
+import org.enso.compiler.core.Persistance;
+import org.enso.compiler.core.ir.expression.Application;
+import org.enso.compiler.core.ir.expression.Case;
+import org.enso.compiler.core.ir.expression.Foreign;
+import org.enso.compiler.core.ir.expression.warnings.Unused;
+import org.enso.compiler.core.ir.module.scope.Definition;
+import org.enso.compiler.core.ir.module.scope.Export;
+import org.enso.compiler.core.ir.module.scope.Import;
+import org.enso.compiler.core.ir.module.scope.definition.Method;
+import org.enso.compiler.core.ir.module.scope.imports.Polyglot;
+import org.enso.compiler.core.ir.type.Set;
+import org.enso.interpreter.dsl.Persistable;
+import org.openide.util.lookup.ServiceProvider;
+import scala.Option;
+import scala.Tuple2;
+import scala.collection.Iterator;
+import scala.collection.SeqFactory;
+import scala.collection.immutable.AbstractSeq;
+import scala.collection.immutable.List;
+import scala.collection.immutable.Seq;
+
+@Persistable(clazz = Polyglot.Java.class, id = 703)
+@Persistable(clazz = DefinitionArgument.Specified.class, id = 704)
+@Persistable(clazz = Name.Self.class, id = 705)
+@Persistable(clazz = Literal.Number.class, id = 706)
+@Persistable(clazz = Literal.Text.class, id = 707)
+@Persistable(clazz = CallArgument.Specified.class, id = 708)
+@Persistable(clazz = Definition.Type.class, id = 709)
+@Persistable(clazz = Definition.Data.class, id = 710)
+@Persistable(clazz = Name.Blank.class, id = 711)
+@Persistable(clazz = Name.GenericAnnotation.class, id = 712)
+@Persistable(clazz = Expression.Block.class, id = 751)
+@Persistable(clazz = Expression.Binding.class, id = 752)
+@Persistable(clazz = Application.Prefix.class, id = 753)
+@Persistable(clazz = Application.Force.class, id = 754)
+@Persistable(clazz = Application.Sequence.class, id = 755)
+@Persistable(clazz = Case.Expr.class, id = 761)
+@Persistable(clazz = Case.Branch.class, id = 762)
+@Persistable(clazz = Pattern.Constructor.class, id = 763)
+@Persistable(clazz = Pattern.Name.class, id = 764)
+@Persistable(clazz = Pattern.Literal.class, id = 765)
+@Persistable(clazz = Pattern.Type.class, id = 766)
+@Persistable(clazz = Method.Conversion.class, id = 771)
+@Persistable(clazz = Set.Union.class, id = 772)
+@Persistable(clazz = Foreign.Definition.class, id = 781)
+@Persistable(clazz = Type.Function.class, id = 782)
+@Persistable(clazz = Name.BuiltinAnnotation.class, id = 783)
+@Persistable(clazz = Type.Error.class, id = 784)
+@Persistable(clazz = Unused.Binding.class, id = 785)
+public final class IrPersistance {
```

Comment:
minor: Is it necessary for all the classes here to be nested classes? What about moving them into a separate package?

Reply:
I reduced the number of classes in 68ed2a7, but otherwise it is probably better to keep them co-located in `IrPersistance` and `PassPersistance`.